source/MaterialXContrib/MaterialXNode/CreateMaterialXNodeCmd.cpp

```diff
@@ -47,46 +46,56 @@ MStatus CreateMaterialXNodeCmd::doIt( const MArgList &args )
 		return status;
 
 	MString materialXDocument;
-	MString elementName;
+	MString elementPath;
 	if( parser.isFlagSet(kDocumentFlag) )
 	{
 		argData.getFlagArgument(kDocumentFlag, 0, materialXDocument);
 	}
 	if (parser.isFlagSet(kElementFlag))
 	{
-		argData.getFlagArgument(kElementFlag, 0, elementName);
+		argData.getFlagArgument(kElementFlag, 0, elementPath);
 	}
-	if (materialXDocument.length() > 0 && elementName.length() > 0)
+	if (materialXDocument.length() > 0 && elementPath.length() > 0)
 	{
-		// Load document
-		MaterialX::DocumentPtr doc = MaterialX::createDocument();
-		MaterialX::readFromXmlFile(doc, materialXDocument.asChar());
-
-		// Load libraries
-		MaterialX::FilePath libSearchPath = Plugin::instance().getLibrarySearchPath();
-		const MaterialX::StringVec libraries = { "stdlib", "pbrlib", "bxdf", "stdlib/genglsl", "pbrlib/genglsl" };
-        MaterialX::loadLibraries(libraries, libSearchPath, doc);
-
-		// Check to make sure the elementName is a valid output node
-		if (!validOutputSpecified(doc, elementName.asChar()))
+		MaterialXData* materialXData = new MaterialXData(materialXDocument.asChar(), elementPath.asChar());
```

Comment:
This should not be in the constructor since we also need to handle when a file get's loaded back in. I was amiss as in that case we must create the document and get the element from the attributes. We naturally don't recreate the input and output attributes. 

If we delete this member and the node is holding on to a reference then the node will be holding on to garbage.
Thus I think it's better to create once and update as needed. If this is already occuring then ignore this :)

Reply:
Can we defer this discussion to a later PR? It's not a trivial and it sounds like this merge is blocking some of you surface work.